controllers/replication-controller/dellcsireplicationgroup_controller.go

```diff
@@ -478,3 +512,274 @@ func (r *ReplicationGroupReconciler) SetupWithManager(mgr ctrl.Manager, limiter
+		return fmt.Errorf("error making target PV %s reclaim policy Retain: %s", pvc.Annotations[controller.RemotePV], err)
+	}
+
+	// Delete the existing PVC
+	err = client.DeletePersistentVolumeClaim(ctx, pvc)
+	if err != nil {
+		return fmt.Errorf("error deleting PVC %s", pvcName)
+	}
+
+	// Wait until PVC is deleted
+	done := false
+	for iteration := 0; !done; iteration++ {
+		time.Sleep(2 * time.Second)
+		_, err := client.GetPersistentVolumeClaim(ctx, namespace, pvcName)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				done = true
+				break
+			}
+
+			return fmt.Errorf("error when waiting for PVC %s/%s to be deleted", namespace, pvcName)
+		}
+
+		if iteration > 30 {
+			return fmt.Errorf("timed out waiting on PVC %s/%s to be deleted", namespace, pvcName)
+		}
+	}
+
+	// Swap some fields in the PVC.
+	localPV := pvc.Spec.VolumeName
+	pvc.Annotations[controller.RemotePV] = pvc.Spec.VolumeName
+	pvc.Spec.VolumeName = targetPV
+
+	remoteStorageClassName := pvc.Annotations[controller.StorageClassRemoteStorageClassParam]
+	pvc.Annotations[controller.StorageClassRemoteStorageClassParam] = *pvc.Spec.StorageClassName
+	pvc.Annotations[controller.ReplicationGroup] = rgTarget
+	pvc.Labels[controller.ReplicationGroup] = rgTarget
+	pvc.Spec.StorageClassName = &remoteStorageClassName
+	pvc.ObjectMeta.ResourceVersion = ""
+
+	// Re-create the PVC, now pointing to the target.
+	err = client.CreatePersistentVolumeClaim(ctx, pvc)
+	if err != nil {
+		return fmt.Errorf("unable to create PVC %s: %s", pvc.Name, err.Error())
+	}
+
+	// Update the target PV's claimref to point to our pvc.
+	pvcUID := pvc.ObjectMeta.UID
+	pvcResourceVersion := pvc.ObjectMeta.ResourceVersion
+	err = updatePVClaimRef(ctx, client, targetPV, pvc.Namespace, pvcResourceVersion, pvc.Name, pvcUID, log)
+	if err != nil {
+		return fmt.Errorf("unable to update PVClaimRef for %s: %s", targetPV, err.Error())
+	}
+
+	// Verify pvc is created and bound to new PVs
+	// remotePV is the current localPVName arg, localPV is the current remotePVName arg
+	err = verifyPVC(ctx, client, targetPV, localPV, pvcName, namespace)
+	if err != nil {
+		return fmt.Errorf("error verifying PVC %s: %s", pvcName, err.Error())
+	}
+
+	// Remove the PVC reclaim of local PV
+	err = removePVClaimRef(ctx, client, localPV, namespace, pvcName, log)
+	if err != nil {
+		return fmt.Errorf("error removing PV claim ref from %s: %s", localPV, err.Error())
+	}
+
+	// Restore the PVs original volume reclaim policy
+	err = setPVReclaimPolicy(ctx, client, pvc.Spec.VolumeName, remotePVPolicy)
+	if err != nil {
+		return fmt.Errorf("restoring PV reclaim policy of %s: %s", pvc.Spec.VolumeName, err.Error())
+	}
+	err = setPVReclaimPolicy(ctx, client, pvc.Annotations[controller.RemotePV], localPVPolicy)
+	if err != nil {
+		return fmt.Errorf("restoring PV reclaim policy of %s: %s", pvc.Annotations[controller.RemotePV], err.Error())
```

Comment:
Can we update the error message to indicate whether it is for local or remote PV?